edgepi-dac.js

```diff
@@ -13,7 +13,10 @@ module.exports = function (RED) {
         node.status({ fill: "green", shape: "dot", text: "input received" });
         try {
           gain = msg.gain || gain;
-          voltage = msg.payload || voltage;
+          voltage =
+            msg.payload && typeof msg.payload == "number"
+              ? msg.payload
+              : voltage;
           channel = msg.channel || channel;

```

Comment:
why do we check just `msg.payload` and not other parts of the message like `msg.gain` and `msg.channel`?

Reply:
Code has been updated. But to answer your question with my old code:

`msg.gain` and `msg.channel` are already checked with || operator